src/main/java/fi/metatavu/kuntaapi/server/controllers/TaskController.java

```diff
@@ -231,47 +157,162 @@ public boolean isQueueExisting(String queueName) {
+    if (e.getCause() instanceof JDBCException) {
+      handleCreateTaskErrorJdbc(queueName, uniqueId, (JDBCException) e.getCause());
+    } else {
+      logger.log(Level.SEVERE, "Task creating failed on unexpected persistence error", e);
+    }
+  }
+
+  private void handleCreateTaskErrorJdbc(String queueName, String uniqueId, JDBCException e) {
+    if (e.getCause() instanceof SQLException) {
+      SQLException sqlException = (SQLException) e.getCause();
+      if (sqlException.getErrorCode() == 1062) {
+        logger.log(Level.SEVERE,  () -> String.format("Task %s insterted twice into queue %s. Skipped", uniqueId, queueName));
+        return;
+      }
+    }
+
+    logger.log(Level.SEVERE, "Task creating failed on unexpected JDBC error", e);
+  }
   
   @SuppressWarnings ("squid:S1168")
-  private <T extends AbstractTask> byte[] serialize(T task) {
+  private <T extends Task> byte[] serialize(T task) {
     try (ByteArrayOutputStream resultStream = new ByteArrayOutputStream()) {
       serializeToStream(task, resultStream);
       resultStream.flush();
       return resultStream.toByteArray();
     } catch (IOException e) {
-      if (logger.isLoggable(Level.SEVERE)) {
-        logger.log(Level.SEVERE, "Failed to write serialized task data", e);
-      }
+      logger.log(Level.SEVERE, "Failed to write serialized task data", e);
     }
     
     return null;
   }
 
-  private <T extends AbstractTask> void serializeToStream(T task, ByteArrayOutputStream resultStream) {
+  private <T extends Task> void serializeToStream(T task, ByteArrayOutputStream resultStream) {
     try (ObjectOutputStream objectStream = new ObjectOutputStream(resultStream)) {
       objectStream.writeObject(task);
       objectStream.flush();
     } catch (IOException e) {
-      if (logger.isLoggable(Level.SEVERE)) {
-        logger.log(Level.SEVERE, "Failed to serialize task", e);
-      }
+      logger.log(Level.SEVERE, "Failed to serialize task", e);
     }
   }
 
-  private <T extends AbstractTask> T unserialize(byte[] rawData) {
+  private <T extends Task> T unserialize(byte[] rawData) {
     try (ByteArrayInputStream byteStream = new ByteArrayInputStream(rawData)) {
       return unserializeFromStream(byteStream);
     } catch (IOException e) {
-      if (logger.isLoggable(Level.SEVERE)) {
-        logger.log(Level.SEVERE, "Failed to write unserialized task data", e);
-      }
+      logger.log(Level.SEVERE, "Failed to write unserialized task data", e);
     }
     
     return null;
   }
 
-  @SuppressWarnings({"unchecked", "squid:S4508"})
-  private <T extends AbstractTask> T unserializeFromStream(ByteArrayInputStream byteStream) {
+  @SuppressWarnings("unchecked")
+  private <T extends Task> T unserializeFromStream(ByteArrayInputStream byteStream) {
     try (ObjectInputStream objectStream = new ObjectInputStream(byteStream)) {
       Object object = objectStream.readObject();
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Make sure deserializing objects is safe here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS4508)